src/main/java/net/vivin/gradle/versioning/VersionComparator.java

```diff
@@ -0,0 +1,81 @@
+package net.vivin.gradle.versioning;
+
+import java.util.Comparator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class VersionComparator implements Comparator<String> {
```

Comment:
Ah, I pulled it out into an own class. Forgot about that. I think I mainly did this for better testability. Because now you can test the ordering deterministically by testing the comparator instead of creating tags and depending on internal logic to get all paths tested.